shell/browser/serial/electron_serial_delegate.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/serial/electron_serial_delegate.h"
+
+#include <utility>
+
+#include "shell/browser/serial/serial_chooser_context.h"
+#include "shell/browser/serial/serial_chooser_controller.h"
+
+namespace electron {
+
+ElectronSerialDelegate::ElectronSerialDelegate() = default;
+
+ElectronSerialDelegate::~ElectronSerialDelegate() = default;
+
+std::unique_ptr<content::SerialChooser> ElectronSerialDelegate::RunChooser(
+    content::RenderFrameHost* frame,
+    std::vector<blink::mojom::SerialPortFilterPtr> filters,
+    content::SerialChooser::Callback callback) {
+  chooser_controller_ = std::make_unique<SerialChooserController>(
```

Comment:
Does the fact that there's only one SerialChooserController at a time mean that multiple WebContents can't have outstanding serial port requests at the same time?

Reply:
I think we are good now as the serial chooser context is tied to the browser context.

Reply:
I'm confused, there's still only one chooser.

What happens if two different WebContents both try to choose a serial port at the same time? Will this `RunChooser` method be called twice, replacing the previous `SerialChooserController`?

Reply:
@nornagon let me test this and see what it does.

Reply:
@nornagon you are correct.  I'll refactor this to handle multiple web contents requesting a serial port.